### PR TITLE
Update Buffer to fix bugs and corner cases

### PR DESCRIPTION
**Pull request by slacrherbst**
This pull request contains the changes in PR #143 and must not be merged until after PR #143 has been merged and tested. We will then regression test the changes in the PR.

In fact it may be confusing to look at the changes before PR #143 has been merged.

### CODE REVIEW

File: include/rogue/protocols/rssi/Header.h [hunk 1]
```diff
@@ -49,7 +49,7 @@ namespace rogue {
             public:
 
                //! Header Size
-               static const uint32_t HeaderSize = 8;
+               static const  int32_t HeaderSize = 8;
```
Comment (ruck314) [inline include/rogue/protocols/rssi/Header.h]: Why changed from uint32 to int32?

Comment (slacrherbst) [reply include/rogue/protocols/rssi/Header.h]: Because it is passed to an arg which takes a relative adjustment: 

 frame->getBuffer(0)->adjustHeader(-rpr::Header::HeaderSize);

Comment (slacrherbst) [reply include/rogue/protocols/rssi/Header.h]: Wanted to avoid singed/unsigned confusion on that line.
